src/include/OSL/Imathx/Imathx.h

```diff
@@ -248,10 +248,11 @@ affineInverse(const Matrix44 &m)
 // at a per function level, so we will resort to disabling optimization
 // for this function.  Not ideal, but this is already a slow path
 // for exceptional situations.
-// GCC could use __attribute__((optimize("-fno-fast-math")))
 // NOTE:  only using "inline" to get ODR (One Definition Rule) behavior
-static inline OSL_HOSTDEVICE Matrix44
-nonAffineInverse(const Matrix44 &source) OSL_CLANG_ATTRIBUTE(optnone)
+static inline OSL_HOSTDEVICE Matrix44 OSL_GNUC_ATTRIBUTE(optimize("O0"))
```

Comment:
O0 basically turns off optimizations for GCC for that one function:  nonAffineInverse.  The comment above explains.  
```
// Avoid potential aliasing issues with OIIO implementation,
// but main purpose is to turn off FMA which can affect
// rounding and order of operations which can affect results
// with near 0 divisor.  Really only an issue when trying to
// match results between LLVM IR based implementation and
// compiler optimized versions which may have optimized
// differently than the LLVM IR version.
// Clang doesn't currently have a way to control fast-math/fp:strict
// at a per function level, so we will resort to disabling optimization
// for this function.  Not ideal, but this is already a slow path
// for exceptional situations.
```

Bottom line was we were getting slightly different results that compounded into tests that are trying to compare between LLVM IR and other compilers.  Would be nice to turn more optimizations on for this function.  You can remove that attribute and see what happens, failures show up in the tests.  Perhaps additional tests results could be added with FMA enabled to avoid those attributes.  Alternatively some (majority of) tests could be altered to avoid the non-affine case and avoid these issue.